account_invoice_base_invoicing_mode/models/sale_order.py

```diff
@@ -0,0 +1,10 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    invoicing_mode = fields.Selection(related="partner_invoice_id.invoicing_mode")
```

Comment:
this field probably should be computed to consider the state of sale.order 
if we have one invoicing_mode, sold stuff confirm pay deliver, and then change mode on partner then mode on sale order also will change
thou this is a theoretical assumption, it could be that  method implemented to avoid these issue